cmd/dev_common.go

```diff
@@ -90,30 +89,29 @@ func addStackRegistryFlag(cmd *cobra.Command, flagVar *string, config *RootComma
 }
 
 func addDevCommonFlags(cmd *cobra.Command, config *devCommonConfig) {
-	projectName, perr := getProjectName(config.RootCommandConfig)
-	if perr != nil {
-		if _, ok := perr.(*NotAnAppsodyProject); ok {
-			// rootConfig.Debug.log("Cannot retrieve the project name - continuing: ", perr)
-		} else {
-			config.Error.logf("Error occurred retrieving project name... exiting: %s", perr)
-			os.Exit(1)
-		}
-	}
-	defaultDepsVolume := projectName + "-deps"
 
 	addNameFlag(cmd, &config.containerName, config.RootCommandConfig)
 	addStackRegistryFlag(cmd, &config.StackRegistry, config.RootCommandConfig)
 	cmd.PersistentFlags().StringVar(&config.dockerNetwork, "network", "", "Specify the network for docker to use.")
-	cmd.PersistentFlags().StringVar(&config.depsVolumeName, "deps-volume", defaultDepsVolume, "Docker volume to use for dependencies. Mounts to APPSODY_DEPS dir.")
 	cmd.PersistentFlags().StringArrayVarP(&config.ports, "publish", "p", nil, "Publish the container's ports to the host. The stack's exposed ports will always be published, but you can publish addition ports or override the host ports with this option.")
 	cmd.PersistentFlags().BoolVarP(&config.publishAllPorts, "publish-all", "P", false, "Publish all exposed ports to random ports")
 	cmd.PersistentFlags().BoolVar(&config.disableWatcher, "no-watcher", false, "Disable file watching, regardless of container environment variable settings.")
 	cmd.PersistentFlags().BoolVarP(&config.interactive, "interactive", "i", false, "Attach STDIN to the container for interactive TTY mode")
 	cmd.PersistentFlags().StringVar(&config.dockerOptions, "docker-options", "", "Specify the docker run options to use.  Value must be in \"\". The following Docker options are not supported:  '--help','-p','--publish-all','-P','-u','-—user','-—name','-—network','-t','-—tty,'—rm','—entrypoint','-v','—volume'.")
-
 }
 
 func commonCmd(config *devCommonConfig, mode string) error {
+	id, err := getIDFromConfig(config.RootCommandConfig)
+	if err != nil {
+		return err
+	}
+	// create id if it does not exist in .appsody-config.yaml and add it to project.yaml
+	if id == "" {
+		err = generateNewProjectAndID(config.RootCommandConfig)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
getIDFromConfig could call generateNewProjectAndID if internally it cannot find it. This would remove this code, and only call once `getIDFromConfig` at line 175.

Reply:
Very good point, done!